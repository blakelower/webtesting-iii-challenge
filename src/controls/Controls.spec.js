// Test away!
import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Controls from "./Controls";
afterEach(rtl.cleanup);

describe("Controls", () => {
  it("should match snapshot", () => {
    const wrapper = rtl.render(<Controls />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
  it("provides buttons to toggle the closed and locked states", () => {
    const { getByText } = rtl.render(<Controls />);
    getByText(/Lock Gate/);
    getByText(/Close Gate/);
  });
  it("the closed toggle button is disabled if the gate is locked", () => {
    const { getByText } = rtl.render(<Controls closed={true} locked={true} />);
    const openButton = getByText(/Open Gate/);
    expect(openButton.disabled).toBeTruthy();
  });
  it("the locked toggle button is disabled if the gate is open", () => {
    const { getByText } = rtl.render(
      <Controls closed={false} locked={false} />
    );
    const lockButton = getByText(/Lock Gate/);
    expect(lockButton.disabled).toBeTruthy();
  });
});
