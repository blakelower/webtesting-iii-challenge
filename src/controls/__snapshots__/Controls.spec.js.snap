// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot 1`] = `
<body>
  <div>
    <div
      class="controls panel"
    >
      <button
        class="toggle-btn"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;
